Platform/CommonBoardPkg/Library/BoardSupportLib/BoardSupportLib.c

```diff
@@ -98,6 +99,17 @@ FillBootOptionListFromCfgData (
       CopyMem (BootOption, BootOptionCfgData, OFFSET_OF (OS_BOOT_OPTION, Image[0]));
     }
 
+    Image = &BootOptionCfgData->Image[0].ContainerImage;
+    if ((UpdateFlag == 2) && (Image->Indicate == '!') && (Image->BackSlash == '/')) {
+      // Currently addendum boot option only support RTCM and preOs
+      if ((Image->ContainerSig == SIGNATURE_32 ('I', 'P', 'F', 'W')) && (Image->ComponentName == SIGNATURE_32 ('T', 'C', 'C', 'R'))) {

```

Comment:
For this function, I think it is better to let the config item to indicate which image type the addendum is.  
So here we don't have to compare the signature.  Also it is not good to assume PreOs image for any addendum entry.

Reply:
Sure. let me update the patch